src/plugins/borrow-plugins/types.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import { type EdgeAccount, type EdgeCurrencyWallet, type EdgeNetworkFee } from 'edge-core-js'
+
+// Proposal for a "service plugin" which is a Higher-Order-Plugin that can be
+// instantiated with some account context and returns a service instance.
+export type EdgeServicePlugin<Info, Api> = {
+  info: Info,
+  init: (account: EdgeAccount) => Promise<Api>
+}
+
+// Borrow more:
+export type BorrowRequest = {
+  tokenId?: string,
+  nativeAmount: string,
+
+  // Optional wallet which will deposit on behalf of the loan account mainWallet
+  fromWallet?: string
+}
+
+// Make payment:
+export type RepayRequest = {
+  tokenId?: string,
+  nativeAmount: string,
+
+  // Optional wallet which will deposit on behalf of the loan account mainWallet
+  fromWallet?: string
+}
+
+// Deposit collateral:
+export type DepositRequest = {
+  tokenId?: string,
+  nativeAmount: string,
+
+  // Optional wallet which will deposit on behalf of the loan account mainWallet
+  fromWallet?: string
+}
+
+// Withdraw collateral:
+export type WithdrawRequest = {
+  tokenId?: string,
+  nativeAmount: string,
+
+  // Optional destination for the funds
+  toWallet?: string
+}
+
+// General purpose approvable action
+export type ApprovableAction = {
+  +networkFee: EdgeNetworkFee,
+  +approve: () => Promise<void>
+}
+
+//
+// Loan Account – Primary entity
+//
+
+export type LoanAccount = {
+  // Account wallet for the loan (e.g. Ethereum wallet)
+  mainWallet: EdgeCurrencyWallet,
+
+  collaterals: Array<{
+    tokenId?: string,
+    nativeAmount: string
+  }>,
+  debts: Array<{
+    tokenId?: string,
+    nativeAmount: string,
+    apy: number
+  }>,
+
+  loanToValue: number,
+
+  getApr: (tokenId?: string) => Promise<number>,
+
+  // Collateral modification
+  deposit: (request: DepositRequest) => Promise<ApprovableAction>,
+  withdraw: (request: WithdrawRequest) => Promise<ApprovableAction>,
+
+  // Debt modification
+  borrow: (request: BorrowRequest) => Promise<ApprovableAction>,
+  repay: (request: RepayRequest) => Promise<ApprovableAction>,
+
+  // Close entire account
+  close: () => Promise<ApprovableAction>
+}
+
+//
+// Plugin Info
+//
+
+export type BorrowPluginInfo = {
+  pluginId: string,
+
+  // Maximum loan-to-value ratio
+  maxLtvRatio: number
+}
+
+//
+// Service
+//
+
+export type BorrowService = {
+  fetchLoanAccount: (wallet: EdgeCurrencyWallet) => Promise<LoanAccount>,
```

Comment:
We use the term `fetch` for network requests. This is why swap plugins use `fetch`. However, here we are just creating objects, so these should be `makeLoanAccount`, or maybe `makeLoanEngine` or `makeBorrowEngine` as other names.

Reply:
These are network request though. Each "fetch" method will read the state on the blockchain.